tests/workflow_tests/test_deployment_workflows.py

```diff
@@ -82,3 +85,37 @@ def test_get_workflow_parameters(self):
             }
         }
         self.assertEqual(expected_params, execute_op_workflow.parameters)
+
+    def test_delete_botched_deployment(self):
+
+        from testenv import testenv_instance
+        storage_file_path = os.path.join(
+            testenv_instance.plugins_storage_dir,
+            'agent.json'
+        )
+
+        dsl_path = resource('dsl/basic.yaml')
+        _id = uuid.uuid1()
+        blueprint_id = 'blueprint_{0}'.format(_id)
+        deployment_id = 'deployment_{0}'.format(_id)
+
+        data = {
+            deployment_id: {'raise_exception_on_delete': True}
+        }
+        with open(storage_file_path, 'w') as f:
+            json.dump(data, f)
+
+        self.client.blueprints.upload(dsl_path, blueprint_id)
+        self.client.deployments.create(blueprint_id, deployment_id)
+
+        # wait for creation to finish
+        from time import sleep
```

Comment:
you have to use sleep cause the creation has to finish before you can delete it
i'll use the spin lock ```wait_for_execution_to_end```